### PR TITLE
Move tutorials, rename docs

### LINKED ISSUE #4569: Make code more discoverable by moving component crates to the top level
I've been browsing a lot of Rust crates over the past few days for a personal project, and have found it super valuable if a crate is easy to locate when following the "Repository" link from docs.rs. This is useful for looking at the implementation, checking out dependencies, and for locating examples. Our crates are currently pretty hard to find, for example for `icu_list` you have to know it's in `components/list`. My proposal is:
* Move all components out of `components` into the top level, into directories that match the crate names. This will make `icu` be sorted first, followed by all `icu_<component>` crates
* Move `experimental/components` to the top level as `icu_experimental`
* Move the remaining crates in `experimental` into either `ffi` or `utils`. The experimental status is not very useful for utils imho, most crates in utils are pre-1.0 and don't fully conform to our standards
* Potentially move client-facing provider crates (_blob, _fs, _adapters, _datagen) to the top level as well
* Maybe move our tutorials from `docs/tutorials` into a top-level `examples` folder

### PR DESCRIPTION
**Pull request by robertbastian**
Fixes #4569

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Review by sffc — APPROVED** [review]

File: .github/workflows/artifacts-build.yml [hunk 1]
```diff
@@ -196,7 +196,7 @@ jobs:
     - name: Upload docs to Google Cloud Storage (main)
       if: github.ref == 'refs/heads/main' && github.repository == 'unicode-org/icu4x'
       run: |
-        gsutil -m cp -r ffi/npm/docs/* gs://${{ env.GCP_MAIN_BUCKET_ID }}/gha/tsdoc
+        gsutil -m cp -r ffi/npm/documents/* gs://${{ env.GCP_MAIN_BUCKET_ID }}/gha/tsdoc
```
Comment (sffc) [inline .github/workflows/artifacts-build.yml:199]: This should still be `docs`

File: CODEOWNERS [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # See https://github.blog/2017-07-06-introducing-code-owners/ for details
-# Format of the file follows: https://git-scm.com/docs/gitattributes
+# Format of the file follows: https://git-scm.com/documents/gitattributes
```
Comment (sffc) [inline CODEOWNERS:2]: `docs`

File: documents/process/prd.md [hunk 3]
```diff
@@ -81,9 +81,9 @@ We identified a number of opportunities to evaluate progress of ICU4X on the pat
 
 With the upcoming 0.2 release, ICU4X will provide at least three components that can be validated against a comprehensive internationalization test harness - [test262](https://github.com/tc39/test262) - used by ECMA-402:
 
--   [Intl.Locale](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale)
--   [Intl.PluralRules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/PluralRules)
--   [Intl.DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat)
+-   [Intl.Locale](https://developer.mozilla.org/en-US/documents/Web/JavaScript/Reference/Global_Objects/Intl/Locale)
+-   [Intl.PluralRules](https://developer.mozilla.org/en-US/documents/Web/JavaScript/Reference/Global_Objects/Intl/PluralRules)
+-   [Intl.DateTimeFormat](https://developer.mozilla.org/en-US/documents/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat)
```
Comment (sffc) [inline documents/process/prd.md:86]: `docs`

File: documents/process/prd.md [hunk 4]
```diff
@@ -189,7 +189,7 @@ On the business side, we get a chance to supply the most common user needs, and
 
 Number formatting represents a blend of low-level foundational APIs such as Locale, Plural Rules and CodePointSet, with high-level APIs such as DateTimeFormat.
 
-It is required for DateTimeFormat, and will exercise our ability to maintain modularity and lean profile of our APIs, while building a large feature set that [ECMA-402 NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) carries.
+It is required for DateTimeFormat, and will exercise our ability to maintain modularity and lean profile of our APIs, while building a large feature set that [ECMA-402 NumberFormat](https://developer.mozilla.org/en-US/documents/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) carries.
```
Comment (sffc) [inline documents/process/prd.md:192]: `docs`

File: documents/process/prd.md [hunk 5]
```diff
@@ -205,7 +205,7 @@ Components listed below are worth considering for 1.0 as resources and timing pe
 
 ### Collator
 
-Collation is one of the core [ECMA-402 components](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator), and builds on top of the Unicode Properties API which we’re investing in.
+Collation is one of the core [ECMA-402 components](https://developer.mozilla.org/en-US/documents/Web/JavaScript/Reference/Global_Objects/Intl/Collator), and builds on top of the Unicode Properties API which we’re investing in.
```
Comment (sffc) [inline documents/process/prd.md:208]: `docs`

File: documents/proposals/pr001-ecma402.md [hunk 6]
```diff
@@ -16,7 +16,7 @@ _Usable text internationalization for Rust, **today**._
 [ECMA402][ecma402] is a state-of-the-art Intl API, Rust needs Intl API,
 designing trait-set for ecma402 surface is the logical next step.
 
-[ecma402]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl
+[ecma402]: https://developer.mozilla.org/en-US/documents/Web/JavaScript/Reference/Global_Objects/Intl
```
Comment (sffc) [inline documents/proposals/pr001-ecma402.md:19]: `docs`

File: documents/proposals/pr002-hunkspace.md [hunk 7]
```diff
@@ -99,7 +99,7 @@ from a place that the software distributor doesn't even know exists.  In the
 simplest form, you can download a language pack from a website of your choice,
 register it with your piece of software and you get a localized version of your
 program. This is approximately how [Mozilla's
-XPI](https://developer.mozilla.org/en-US/docs/Mozilla/XPI) works with
+XPI](https://developer.mozilla.org/en-US/documents/Mozilla/XPI) works with
```
Comment (sffc) [inline documents/proposals/pr002-hunkspace.md:102]: `docs`

File: documents/research/datetime.md [hunk 8]
```diff
@@ -1,10 +1,10 @@
 # Rust Date and Time API Request
 
-In ICU4X 0.1 we introduce a `DateTimeFormat` API for formatting date and time into human readable formats. That API is shaped after [`ICU4C`](http://userguide.icu-project.org/formatparse/datetime), [`ICU4J`](https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/text/DateFormat.html) and [`ECMA-402`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat).
+In ICU4X 0.1 we introduce a `DateTimeFormat` API for formatting date and time into human readable formats. That API is shaped after [`ICU4C`](http://userguide.icu-project.org/formatparse/datetime), [`ICU4J`](https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/text/DateFormat.html) and [`ECMA-402`](https://developer.mozilla.org/en-US/documents/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat).
```
Comment (sffc) [inline documents/research/datetime.md:3]: `docs`

A lot of these are on developer.mozilla.org. I won't leave any more comments for these. Please do a global search and replace.

File: documents/research/datetime.md [hunk 9]
```diff
@@ -18,14 +18,14 @@ In ICU4X, we also intend to implement a low-level, performance-focused API for c
...
 
 In particular, we believe that correct date and time management API must include `calendar` and `time zone`.
 It is our experience that such functionality is incredibly hard to retrofit into an existing API. Therefore it is our belief that such functionality must be planned very early in the API design.
 We do not see any crate in the current ecosystem that covers those two features.
 
 ## Temporal
 
-One example of an API that is being meticulously designed to handle the scope correctly is [EcmaScript Temporal Proposal](https://tc39.es/proposal-temporal/docs/index.html). We encourage everyone interested in the topic to read through the [Design Doc](https://github.com/tc39/proposal-temporal/blob/main/docs/calendar-draft.md) for it.
+One example of an API that is being meticulously designed to handle the scope correctly is [EcmaScript Temporal Proposal](https://tc39.es/proposal-temporal/documents/index.html). We encourage everyone interested in the topic to read through the [Design Doc](https://github.com/tc39/proposal-temporal/blob/main/documents/calendar-draft.md) for it.
```
Comment (sffc) [inline documents/research/datetime.md:29]: `docs`

**Review by sffc — CHANGES_REQUESTED** [review]
Looks like `documents` to `docs` was a bit too aggressive

**Comment by sffc** [comment]
Missed a few: https://github.com/unicode-org/icu4x/pull/4734

**Comment by sffc** [comment]
Also broke WASM demo: https://github.com/unicode-org/icu4x/pull/4737

**Pull request merged**